platform-sdk/docs/proposals/template-2.md

```diff
@@ -0,0 +1,184 @@
+
+Remove this section if not applicable.
+
+### Non-Goals
+
+Describe what the proposal is not trying to achieve. What are the limitations of the proposal?
+
+Remove this section if not applicable.
+
+### Dependencies, Interactions, and Implications
+
+What are the dependencies and pre-requisites of this proposal?  
+What other proposals, capabilities, and teams are impacted by this proposal?
+What does the proposal enable once it is completed?
+
+Remove this section if not applicable.
+
+### Requirements
+
+Describe the requirements and acceptance criteria that the design proposal must satisfy.
+
+Remove this section if not applicable.
+
+## Design & Architecture
+
+This section should give an answer to the question: "How will this proposal be implemented?"
+
+This is one of the core sections of the proposal and should be detailed.
+All subtitled sections are optional and can be removed if not applicable.
+Please structure the content in a way that is easy to read and understand for a broad audience.
+A minimal proposal must have a "Design & Architecture" section without any sub-sections.
+In any case the section must give a detailed answer to the question: "How will this proposal be implemented?"
+
+### Architecture and/or Components
+
+Describe any new or modified components or architectural changes. This includes thread management changes, state
+changes, disk I/O changes, platform wiring changes, etc. Include diagrams of architecture changes.
+
+Remove this section if not applicable.
+
+### Module Organization and Repositories
+
+Describe any new or modified modules or repositories.
+
+Remove this section if not applicable.
+
+### Core Behaviors
+
+Describe any new or modified behavior. What are the new or modified algorithms and protocols? Include any diagrams that
+help explain the behavior.
+
+Remove this section if not applicable.
+
+### Public API
+
+Describe any public API changes or additions. Include stakeholders of the API.
+
+Examples of public API include:
+
+* Anything defined in protobuf
+* Any functional API that is available for use outside the module that provides it.
+* Anything written or read from disk
+
+Code can be included in the proposal directory, but not committed to the code base.
+
+Remove this section if not applicable.
+
+### Configuration
+
+Describe any new or modified configuration.
+
+Remove this section if not applicable.
+
+### Metrics
+
+Are there new metrics? Are the computation of existing metrics changing? Are there expected observable metric impacts
+that change how someone should relate to the metric?
+
+Remove this section if not applicable.
+
+### Performance
+
+Describe any expected performance impacts. This section is mandatory for platform wiring changes.
+
+Remove this section if not applicable.
+
+### Migration
+
+Describe the migration path for existing usage.
+What changes are needed to upgrade from the current state to the new state?
+
+Remove this section if not applicable.
+
+## Testing
```

Comment:
I actually prefer "Test Plan", because it indicates a purposeful set of actions that have been set out ahead of time. But the exact wording is not super important.

Reply:
Fine to change it. For me "Test Plan" is way more concrete than what I would expect in that section.
Like "This is the exact plan/schedule of tests that must be executed"